stream/stream_smb2.c

```diff
@@ -0,0 +1,440 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "config.h"
+
+#include <unistd.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <netdb.h>
+#include <poll.h>
+#include <smb2/smb2.h>
+#include <smb2/libsmb2.h>
+#include <smb2/libsmb2-raw.h>
+
+#include "common/msg.h"
+#include "stream.h"
+#include "options/m_option.h"
+
+#include "config.h"
+
+static pthread_mutex_t smb2_lock = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
Does it really need a central lock? I doubt this. The mpv core never accesses the same stream in a concurrent way. It seems like this mutex is copied and cargo-culted from stream_smb.c, which uses a library which isn't library-safe.